packages/backend/src/models/SpaceModel.ts

```diff
@@ -705,7 +713,10 @@ export class SpaceModel {
 
                 // exclude all users that were converted to organization members and have no space access
                 if (!highestRole) {
-                    this.removeSpaceAccess(spaceUuid, user_uuid); // remove access from the space if it exists
+                    // revoke direct access from the space
+                    if (user_with_direct_access) {
+                        this.removeSpaceAccess(spaceUuid, user_uuid);
```

Comment:
Related to this right? `amend logic to only trigger delete query when needed`

Reply:
This is trying to keep the db clean when users no longer have direct access to a space.
Example:
 - user A is an org editor
 - user A is given access to a private space in project X
 - user A is changed to org member and has no access to the project X

Reply:
Removed logic.